src/view/dataElements/xdmObject.jsx

```diff
@@ -79,45 +113,102 @@ const XdmObject = ({ initInfo, formikProps, schema }) => {
+        return fetchSchemaList({
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
-        }).then(fetchedSchema => {
-          setSchema(fetchedSchema);
+        })
+          .then(_schemasMeta => {
+            setSchemasMeta(_schemasMeta);
 
-          if (!fetchedSchema) {
-            return {};
-          }
+            if (!_schemasMeta.length) {
+              return {};
+            }
 
-          const initialValues = getInitialValues({
-            settings: initInfo.settings,
-            schema: fetchedSchema
-          });
+            // TODO: don't initialize form state if the schema doesn't match
+            let existingSchema;
+            if (initInfo.settings && initInfo.settings.schema) {
+              existingSchema = findSchemaMetaById({
+                $id: initInfo.settings.schema.id,
+                schemasMeta: _schemasMeta
+              });
+            }
+
+            let schemaMeta = _schemasMeta[0];
+            if (existingSchema) {
+              schemaMeta = existingSchema;
+            }
+
+            setSelectedSchemaMeta(schemaMeta);
 
-          return initialValues;
-        });
+            return fetchSchema({
+              orgId: initInfo.company.orgId,
+              imsAccess: initInfo.tokens.imsAccess,
+              schemaMeta
+            });
+          })
+          .then(schema => {
+            const initialValues = getInitialFormState({
+              value: (initInfo.settings && initInfo.settings.data) || {},
+              schema
+            });
+
+            return initialValues;
+          });
       }}
       getSettings={({ values }) => {
         return {
           schema: {
-            id: schema.$id,
-            version: schema.version
+            id: selectedSchemaMeta.$id,
+            version: selectedSchemaMeta.version
           },
           data: getValueFromFormState({ formStateNode: values }) || {}
         };
       }}
       validate={validate}
-      render={props => <XdmObject {...props} schema={schema} />}
+      render={options => {
+        const onSchemaMetaSelected = _schemaMeta => {
+          const schemaMeta = findSchemaMetaById({
+            $id: _schemaMeta.$id,
+            schemasMeta
+          });
+          setSelectedSchemaMeta(schemaMeta);
+          fetchSchema({
+            orgId: options.initInfo.company.orgId,
+            imsAccess: options.initInfo.tokens.imsAccess,
+            schemaMeta
+          }).then(schema => {
+            const initialValues = getInitialFormState({
+              value: {},
+              schema
+            });
+            options.resetForm(initialValues);
+          });
+        };
+
+        return (
+          <XdmObject
+            {...options}
+            schemasMeta={schemasMeta}
+            selectedSchemaMeta={selectedSchemaMeta}
+            setSelectedSchemaMeta={onSchemaMetaSelected}
+          />
+        );
+      }}
```

Comment:
I haven't thought this through much, but it looks like it's about time to start breaking this module down into smaller modules. This isn't something I would recommend doing before GA, but maybe after.